public/models.py

```diff
@@ -958,6 +967,37 @@ def get_context(self, request):
         except AttributeError:
             building_str = None
 
+        def next_unit(unit):
+            if unit.title == "Units":
```

Comment:
So Irina and I played around with this, but it looks like successive calls to `get_parent()` lead us to one of these:
```
<class 'wagtail.core.models.Page'>
```
...rather than one of these:
```
<class 'units.models.UnitIndexPage'>
```
I'm guessing this is due to some OO inheritance jazz.  But unfortunately, stopping on a `wagtail.core.models.Page` means stopping too early.  I really like the idea of checking something more permanent than the title, but I'm not sure what the best option is.  What do you think?

Reply:
You have to add the `specific` to the call. That will give you the actual page subtype.

Reply:
Wow, didn't know about that `specific` method; thanks for the background.  Absolutely wild.  Why on earth is Wagtail set up that way?  I can't think of any reason you wouldn't want to always just return the most specific class in the class hierarchy.

Reply:
There are lots of times you don't want the specific page type. For example, if you want all pages on the site, you can't query by specific page type or assume page specific variables/methods are present. You also might have code that needs to work with various page types but not all page types, stuff like that. Wagtail can't assume it knows what context you're in. Also, all of Wagtail's built in functionality has to operate using the core `Page` type to the extant that's applicable.

Reply:
`specific` is currently working great on `nest`!  w00t; excellent idea.

Reply:
Re: the motivation behind `specific`, hmm; still feels weird to me since there's inheritance and whatnot, but I'd have to look at a specific example to articulate why in detail.  Maybe I'll wait until an example like that comes up in the wild.